code/game/machinery/ammolathe.dm

```diff
@@ -0,0 +1,64 @@
+/obj/machinery/r_n_d/fabricator/mechanic_fab/autolathe/ammolathe
+	name = "\improper Ammolathe"
+	desc = "Produces guns, ammunition, and firearm accessories."
+	icon_state = "ammolathe"
+	icon_state_open = "ammolathe_t"
+	nano_file = "ammolathe.tmpl"
+	default_mat_overlays = TRUE
+
+	start_end_anims = 1
+
+	build_time = 0.5
+
+	allowed_materials = 0
+
+	machine_flags = SCREWTOGGLE | CROWDESTROY | WRENCHMOVE | FIXED2WORK //| EMAGGABLE
+
+	research_flags = NANOTOUCH | TAKESMATIN | HASOUTPUT | IGNORE_CHEMS | HASMAT_OVER | LOCKBOXES
+
+	light_color = LIGHT_COLOR_RED
+
+	one_part_set_only = 0
+	part_sets = list(
+		"Weapons"=list(
+		new /obj/item/weapon/gun/projectile/sec/lockbox(), \
+		new /obj/item/weapon/gun/projectile/automatic/vector/lockbox(), \
+		),
+		"Single_ammunition"=list(
+		new /obj/item/ammo_casing/shotgun/flare(), \
+		new /obj/item/ammo_casing/shotgun/beanbag(), \
+		new /obj/item/ammo_casing/shotgun/stunshell(), \
+		new /obj/item/ammo_casing/shotgun(), \
+		new /obj/item/ammo_casing/shotgun/buckshot(),\
+		),
+		"Box_ammunition"=list(
+		new /obj/item/ammo_storage/box/b380auto(), \
+		new /obj/item/ammo_storage/box/b380auto/practice(), \
+		new /obj/item/ammo_storage/box/b380auto/rubber(), \
+		),
+		"Magazines"=list(
+		new /obj/item/ammo_storage/magazine/smg9mm/empty(), \
+		new /obj/item/ammo_storage/magazine/beretta/empty(), \
+		new /obj/item/ammo_storage/magazine/a12mm/empty(), \
+		new /obj/item/ammo_storage/magazine/a357/empty(),\
+		new /obj/item/ammo_storage/magazine/m380auto/empty(), \
+		new /obj/item/ammo_storage/magazine/c45/empty(), \
+		new /obj/item/ammo_storage/magazine/uzi45/empty(), \
+		new /obj/item/ammo_storage/magazine/a50/empty(), \
+		new /obj/item/ammo_storage/magazine/a75/empty(), \
+		new /obj/item/ammo_storage/magazine/a762/empty(), \
+		new /obj/item/ammo_storage/magazine/a12ga/empty(), \
+		),
+		"Misc_Other"=list(
+		new /obj/item/ammo_storage/speedloader/c38/empty(), \
+		new /obj/item/ammo_storage/speedloader/a357/empty(), \
+		new /obj/item/ammo_storage/speedloader/a762x55/empty(), \
+		new /obj/item/ammo_storage/speedloader/shotgun(), \
+		),
+		"Hidden_Items" = list(
```

Comment:
These suck and turn it from ammolathe to seclathe.

Reply:
Alright I can remove the stuff from the contraband list since a few people have had an issue with it. Any suggestions on what to put in there that doesn't overlap with the above. This means stuff that's not box/single ammo, a gun, etc. I was thinking something like syringes and vials since those are technically ammo for lesser known arms.

Reply:
Item list changed but this didn't close.

Reply:
I thought it would be fitting enough for a hidden option. I'm for leaving it in there but if some other people want it gone then I will cave.

Reply:
The knife gun apparently doesn't exist [yet] so they have been removed.